app/services/reporting/account_reuse_report.rb

```diff
@@ -39,77 +63,266 @@ def account_reuse_emailable_report
+        super(detail_rows:)
+      end
+
+      def organize_results(all_results, idv_results, entity_type)
+        idv_results.each do |result|
+          entity_num = result["num_#{entity_type}"]
+          self.detail_rows[entity_num] =
+            ReuseDetailRow.new.update_details(
+              num_entities: entity_num,
+              num_idv_users: result['num_idv_users'],
+            )
+        end
+
+        all_results.each do |result|
+          entity_num = result["num_#{entity_type}"]
+          if self.detail_rows[entity_num].is_a?(Struct)
+            self.detail_rows[entity_num].update_details(num_all_users: result['num_all_users'])
+          else
+            self.detail_rows[entity_num] =
+              ReuseDetailRow.new.update_details(
+                num_entities: entity_num,
+                num_all_users: result['num_all_users'],
+              )
+          end
+        end
 
-      if !reuse_stats.empty?
-        reuse_stats.each do |result_entry|
-          reuse_total_users += result_entry['num_users']
+        if self.detail_rows.length > 1
+          # If there are results, then remove the zero placeholder
+          self.detail_rows[0] = nil
+          self.detail_rows = self.detail_rows.compact
         end
+      end
+
+      self
+    end
+
+    ReuseSummaryRow = Struct.new(
+      :results, :total_all_users, :total_all_percent, :total_idv_users,
+      :total_idv_percent
+    ) do
+      def initialize(total_all_users: 0, total_all_percent: 0, total_idv_users: 0,
+                     total_idv_percent: 0)
+        super(total_all_users:, total_all_percent:, total_idv_users:, total_idv_percent:)
+      end
 
-        if total_proofed > 0
-          reuse_stats.each_with_index do |result_entry, index|
-            reuse_stats[index]['percentage'] = result_entry['num_users'] / total_proofed.to_f
+      def update_from_results(results, total_proofed)
+        results.each do |result_entry|
+          self.total_all_users += result_entry.num_all_users
+          self.total_idv_users += result_entry.num_idv_users
+        end
 
-            reuse_total_percentage += reuse_stats[index]['percentage']
+        if !results.nil? && !results.empty?
+          if total_proofed > 0
+            # Calculate percentages for breakdowns with both sps and angencies
+            results.each_with_index do |result_entry, index|
+              results[index][:all_percent] =
+                result_entry[:num_all_users] / total_proofed.to_f
+              results[index][:idv_percent] =
+                result_entry[:num_idv_users] / total_proofed.to_f
+
+              self.total_all_percent += results[index][:all_percent]
+              self.total_idv_percent += results[index][:idv_percent]
+            end
           end
         end
+        self.results = results
+
+        self
       end
+    end
+
+    def total_reuse_report
+      return @total_reuse_report if defined?(@total_reuse_report)
+
+      reuse_results = {
```

Comment:
we probably don't need this hash anymore?

Reply:
Ah - good point - I'll remove it.